release.nix

```diff
@@ -11,76 +11,71 @@
+              haskellPackagesNew.callPackage ./nix/cabal-doctest.nix { };
 
-              insert-ordered-containers =
-                haskellPackagesNew.callPackage ./nix/insert-ordered-containers.nix { };
+            insert-ordered-containers =
+              haskellPackagesNew.callPackage ./nix/insert-ordered-containers.nix { };
 
-              neat-interpolation =
-                haskellPackagesNew.callPackage ./nix/neat-interpolation.nix { };
+            neat-interpolation =
+              haskellPackagesNew.callPackage ./nix/neat-interpolation.nix { };
 
-              optparse-applicative =
-                haskellPackagesNew.callPackage ./nix/optparse-applicative.nix { } ;
+            optparse-applicative =
+              haskellPackagesNew.callPackage ./nix/optparse-applicative.nix { } ;
 
-              optparse-generic =
-                haskellPackagesNew.callPackage ./nix/optparse-generic.nix { } ;
+            optparse-generic =
+              haskellPackagesNew.callPackage ./nix/optparse-generic.nix { } ;
 
-              proto3-suite-no-tests =
-                pkgs.haskell.lib.dontCheck
-                  (haskellPackagesNew.callPackage ./default.nix { });
+            proto3-suite-no-tests =
+              pkgs.haskell.lib.dontCheck
+                (haskellPackagesNew.callPackage ./default.nix { });
 
-              proto3-suite =
-                pkgs.haskell.lib.overrideCabal
-                  (haskellPackagesNew.callPackage ./default.nix { })
-                  (oldAttrs: {
-                      patches = [ tests/tests.patch ];
+            proto3-suite =
+              pkgs.haskell.lib.overrideCabal
+                (haskellPackagesNew.callPackage ./default.nix { })
+                (oldAttrs: {
+                    patches = [ tests/tests.patch ];
 
-                      postPatch = ''
-                        substituteInPlace tests/encode.sh --replace @ghc@ ${pkgs.ghc} --replace @bash@ ${pkgs.bash}
-                        substituteInPlace tests/decode.sh --replace @ghc@ ${pkgs.ghc} --replace @bash@ ${pkgs.bash}
-                      '';
+                    postPatch = ''
+                      substituteInPlace tests/encode.sh --replace @ghc@ ${pkgs.ghc} --replace @bash@ ${pkgs.bash}
+                      substituteInPlace tests/decode.sh --replace @ghc@ ${pkgs.ghc} --replace @bash@ ${pkgs.bash}
+                    '';
 
-                      testHaskellDepends = oldAttrs.testHaskellDepends ++ [
-                        pkgs.ghc
-                        proto3-suite-no-tests
-                        pkgs.protobuf3_0
-                        pkgs.python
-                        python_protobuf3_0
-                      ];
-                    }
-                  );
+                    testHaskellDepends = oldAttrs.testHaskellDepends ++ [
+                      pkgs.ghc
+                      proto3-suite-no-tests
+                      pkgs.protobuf3_1
+                      pkgs.python
```

Comment:
I think that `pkgs.python` should be `(pkgs.python.withPackages (pkgs: [ pkgs.protobuf ]))` so that the Python `protobuf` library is in scope for the `python` interpreter.  Otherwise I think that the tests that compare our output to Python output will fail

Reply:
The tests are still broken with 
```
    generate code for a simple .proto and then use it to encode messages:   FAIL
      Exception: user error (PYTHONPATH environment variable is not set)
    generate code for a simple .proto and then use it to decode messages:    FAIL
      Exception: user error (PYTHONPATH environment variable is not set)
```

after this modification.

Reply:
Try using `pythonFull.withPackages` instead of `python.withpackages`.  See: https://github.com/NixOS/nixpkgs/issues/2411

Reply:
That did not fix it -- perhaps I need to override `stdenv` and export the variable there?

Reply:
You don't need to override the `stdenv`.  An easier way to provide the variable is to just add it as a field of the Haskell derivation.  i.e.:

```nix
            pythonWithProtobuf = pkgs.python.withPackages (pkgs: [ pkgs.protobuf ]);

            proto3-suite =
              pkgs.haskell.lib.overrideCabal
                (haskellPackagesNew.callPackage ./default.nix { })
                (oldAttrs: {
                    PYTHONPATH = "${pythonWithProtobuf}/some/path/I/don't/know/exactly/which/one";  # ←

                    patches = [ tests/tests.patch ];

                    postPatch = ''
                      substituteInPlace tests/encode.sh --replace @ghc@ ${pkgs.ghc} --replace @bash@ ${pkgs.bash}
                      substituteInPlace tests/decode.sh --replace @ghc@ ${pkgs.ghc} --replace @bash@ ${pkgs.bash}
                    '';

                    testHaskellDepends = oldAttrs.testHaskellDepends ++ [
                      pkgs.ghc
                      proto3-suite-no-tests
                      pkgs.protobuf3_1
                      pythonWithProtobuf
                    ];
                  }
                );
```

Nix turns all fields of a derivation into environment variables for the derivation (after coercing them into strings if they aren't strings already).

Reply:
I think `PYTHONPATH` just comes in empty now, and that the error is overly aggressive. A quick workaround is to hack the test to append the tmp dir location. There's a lot of cleanup I'm in the middle of, but that should unblock you, @themattchan.

Reply:
@Gabriel439, adding `PYTHONPATH` gave the following error:

```error: anonymous function at /nix/store/igjvsvn1fc1llppp8ml7b6iyp40d4qmb-nixpkgs-fd0abdcebc121c8685e3f1e514f022c59a81b2b5-src/pkgs/development/haskell-modules/generic-builder.nix:6:1 called with unexpected argument ‘PYTHONPATH’, at /nix/store/igjvsvn1fc1llppp8ml7b6iyp40d4qmb-nixpkgs-fd0abdcebc121c8685e3f1e514f022c59a81b2b5-src/lib/customisation.nix:74:12```

Reply:
Oh, oops!  In that case try doing @joel's suggestion and just modify the Haskell test code to not expect a pre-existing `PYTHONPATH`